backend/src/LibExecution/Interpreter.fs

```diff
@@ -628,14 +643,17 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
     | EEnum(id, typeName, caseName, fields) ->
       let typeStr = TypeName.toString typeName
       let types = ExecutionState.availableTypes state
-      match! Types.find typeName types with
-      | None -> return err id $"There is no type named {typeStr}"
-      | Some({ definition = TypeDeclaration.Alias _ }) ->
-        return err id $"Expected an enum but {typeStr} is an alias"
-      | Some({ definition = TypeDeclaration.Record _ }) ->
-        return err id $"Expected an enum but {typeStr} is a record"
-      | Some({ definition = TypeDeclaration.Enum(case, cases) }) ->
-        let case = (case :: cases) |> List.tryFind (fun c -> c.name = caseName)
+      let! typ = Types.find typeName types
+
+      match! enumMaybe types typeName with
+      | None ->
+        match typ with
+        | None -> return err id $"There is no type named {typeStr}"
+        | Some({ definition = TypeDeclaration.Enum _ }) ->
+          return err id $"Expected a record but {typeStr} is an enum"
+        | _ -> return err id $"Expected a record but {typeStr} is something else"
```

Comment:
we might as well exhaust the `match` cases here since there's just Alias and Record remaining? (helpful to catch something later)

Reply:
I'm continuing to do work on it, so I'll do this as part of it. (TypeArgs aren't handled anywhere here)